packages/vite/plugins/nx-tsconfig-paths.plugin.ts

```diff
@@ -0,0 +1,89 @@
+import { stripIndents, workspaceRoot } from '@nx/devkit';
+import { existsSync } from 'node:fs';
+import { relative, join, resolve } from 'node:path';
+import { loadConfig, createMatchPath, MatchPath } from 'tsconfig-paths';
+
+export function nxViteTsPaths() {
+  let matchTsPathEsm: MatchPath;
+  let matchTsPathFallback: MatchPath | undefined;
+
+  return {
+    name: 'nx-vite-ts-paths',
+    configResolved(config: any) {
+      const projectRoot = config.root;
+      const projectRootFromWorkspaceRoot = relative(workspaceRoot, projectRoot);
+
+      const foundTsConfigPath = getTsConfig(
+        join(
+          workspaceRoot,
+          'tmp',
+          projectRootFromWorkspaceRoot,
+          'tsconfig.generated.json'
+        )
+      );
+      if (!foundTsConfigPath) {
+        throw new Error(stripIndents`Unable to find a tsconfig in the workspace! 
+There should at least be a tsconfig.base.json or tsconfig.json in the root of the workspace ${workspaceRoot}`);
+      }
+      const parsed = loadConfig(foundTsConfigPath);
+
+      logIt('first parsed tsconfig: ', parsed);
+      if (parsed.resultType === 'failed') {
+        throw new Error(`Failed loading tsonfig at ${foundTsConfigPath}`);
+      }
+
+      matchTsPathEsm = createMatchPath(parsed.absoluteBaseUrl, parsed.paths, [
+        ['exports', '.', 'import'],
+        'module',
+        'main',
+      ]);
+
+      const rootLevelTsConfig = getTsConfig(
+        join(workspaceRoot, 'tsconfig.base.json')
+      );
+      const rootLevelParsed = loadConfig(rootLevelTsConfig);
+      logIt('fallback parsed tsconfig: ', rootLevelParsed);
+      if (rootLevelParsed.resultType === 'success') {
+        matchTsPathFallback = createMatchPath(
+          rootLevelParsed.absoluteBaseUrl,
+          rootLevelParsed.paths,
+          ['main', 'module']
+        );
+      }
+    },
+    resolveId(source: string) {
+      let resolvedFile: string;
+      try {
+        resolvedFile = matchTsPathEsm(source);
+      } catch (e) {
+        logIt('Using fallback path matching.');
+        resolvedFile = matchTsPathFallback?.(source);
+      }
```

Comment:
This solves the issues I came across where a default @nx/js library will be built with the following package.json
```json
{
  "name": "@acme/js-lib",
  "version": "0.0.1",
  "type": "module",
  "peerDependencies": {
    "tslib": "2.6.0"
  },
  "main": "./src/index.js",
  "types": "./src/index.d.ts"
}
```
and vite will be unable to load it as RollUp claims it cannot find the named exports, or the `tsconfig-paths` will not be able to resolve the exports object since it doesn't exist.

The same issue happens in the original plugin being used as well `vite-tsconfig-paths`. 

This fallbacks to the root level tsconfig where the `.ts` files will be loaded instead so projects that don't include `export` objects or commonjs will still be able to be used in the project with `buildLibsFromSource=false` just will be forced to be build from source. 

This can be fixed in the userland by changing the tsconfig settings to use NodeNext and `type:module` in package.json which allows vite to see the named exports

i.e. the output switches from 

```js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
tslib_1.__exportStar(require("./lib/lib-js"), exports);
//# sourceMappingURL=index.js.map
```
to just

```js
export * from './lib/lib-js';
```

Reply:
@jaysoo would like your opinion on this, to see if this makes sense or if we should see what we can do to get cjs built libs working inside vite.